client/src/components/Friends/AddFriendComponent.js

```diff
@@ -0,0 +1,290 @@
+  noResults: {
+    textAlign: 'center',
+    color: 'gray'
+  },
+  avatar: {
+    border: '1px solid black'
+  },
+  adornedEnd: {
+    paddingRight: 0
+  },
+  dialog: {
+    margin: theme.spacing(1)
+  }
+}))
+
+const AddFriendComponent = ({
+  accountValues,
+  relations,
+  relationsUsers,
+  setSnackbarOpen,
+  setSnackBarMessage,
+  reloadRelations,
+  handleBlacklistModalOpen
+}) => {
+  const classes = useStyles()
+
+  const [searchValue, setSearchValue] = useState('')
+  const [searchResults, setSearchResults] = useState([])
+
+  // Use this to filter out user ids from being able to be searched as part of blacklist
+  const [blockedIds, setBlockedIds] = useState([])
+
+  // Set boolean variable for checking if search results are loading
+  const [bLoading, setBLoading] = useState(false)
+  // Set boolean variable to display "No results" only after making at least one search
+  const [bSearched, setBSearched] = useState(false)
+
+  // Set boolean variable for opening modal
+  const [bModalOpen, setBModalOpen] = useState(false)
+  // User that is being sent a friend request
+  const [friendRequestUser, setFriendRequestUser] = useState({})
+  // Friend Request message
+  const [friendRequestMessage, setFriendRequestMessage] = useState(
+    "Let's be friends! ^-^"
+  )
+
+  const handleKeyPress = e => {
+    if (e.key === 'Enter' || e.key === 'NumpadEnter') {
+      handleSearchSubmit()
+    }
+  }
+
+  const handleSearchValueChange = event => {
+    setSearchValue(event.target.value)
+  }
+
+  const handleSearchSubmit = () => {
+    if (searchValue.length > 0) {
+      setBLoading(true)
+
+      axios
+        .get(`/users/find/${searchValue}`)
+        .then(res => {
+          setSearchResults(res.data)
+        })
+        .catch(err => {
+          console.log(err)
+        })
+        .finally(() => {
+          if (!bSearched) setBSearched(true)
+          setBLoading(false)
+        })
+    }
+  }
+
+  const handleFriendRequestModalOpen = user => {
+    // Set friend request user to selected
+    setFriendRequestUser(user)
+    setBModalOpen(true)
+  }
+
+  const handleFriendRequestModalClose = () => {
+    setBModalOpen(false)
+    // Reset friend request message and user
+    // Added a small delay so you do not see the values resetting as the modal is fading
+    setTimeout(() => {
+      setFriendRequestMessage("Let's be friends! ^-^")
+      setFriendRequestUser({})
+    }, 500)
+  }
+
+  const handleFriendRequestMessageChange = event => {
+    setFriendRequestMessage(event.target.value)
+  }
+
+  const handleFriendRequestSubmit = () => {
+    if (searchValue.length > 0) {
+      axios
+        .post('/users/add/', {
+          // userId1 uses .id because the underline is removed when initially storing (and then retrieving) this data to/from the cookie
+          // userId2 uses ._id as it is directly retrieved from MongoDB
+          userId1: accountValues.id,
+          userId2: friendRequestUser._id,
```

Comment:
Best practice wise, these should probably be named better. Like `sender`, `receiver`

Reply:
Changed to sender/receiver for client, but leaving userId1/userId2 for server because that is how the values are stored in the database (you cannot differentiate sender/receiver because it is sorted by lower id as userId1)